pennylane/operation.py

```diff
@@ -461,6 +462,11 @@ def check_domain(self, p, flattened=False):
                     raise TypeError(
                         "{}: Natural number parameter expected, got {}.".format(self.name, p)
                     )
+        elif self.par_domain == "L":
+            if not isinstance(p, list):
+                raise TypeError("{}: List parameter expected, got {}.".format(self.name, type(p)))
+            if not all(isinstance(elem, np.ndarray) for elem in p):
+                raise TypeError("List elements must be Numpy arrays.")
```

Comment:
A new branch will be needed in the `parameters` property as well for `list` objects. This part failed previously for numpy arrays, now the lists can include numpy arrays which have `Variable` instances in them. This would include something like for the numpy array branch, just iterating through each array:
```python
            if isinstance(p, list):
                # p is assumed to be a list of numpy arrays
                # object arrays may have Variables inside them
                evaled_list = []
                for arr in p:
                    if p.dtype == object:
                        temp.append([x.val if isinstance(x, Variable) else x for x in arr])
                    else:
                        temp.append(p)
                return p
```
Would still be curious to hear @josh146's thoughts if this is going to be fine for trainable parameters.

Reply:
@antalszava should this not go in `evaluate` function ?